### DISCUSSION TITLE
No vertical in-page navigation

### DISCUSSION THREAD

**Discussion by charleneyjh1** [twilio-labs/paste#3307] [Q&A] [answered]
Hello team, 

Per paste guidance we should be using in-page navigation instead of tabs if we are navigating user to a different URL per tab. 
For Segment-> Unify settings we need both a horizontal in-page navigation and a vertical in-page navigation. However it seems like we don't have a vertical in-page navigation per my ENG team's investigation. Could you confirm if that's true?

If that's the case, should we use vertical tabs instead or use 2 horizontal in-page navigation? I personally think two horizontal in-page navigation is bad UX. 

Thank you so much for your help!

<img width="1359" alt="Screenshot 2023-07-05 at 1 38 05 PM" src="https://github.com/twilio-labs/paste/assets/56697620/3bd9ff21-4eb6-4677-ac46-75f385d678ed">

**Comment by SiTaggart** [comment]
Hi @charleneyjh1 

Right now, that is correct, yes. We were likely not presented a use case for vertical in page navigation, as folks didn't tend to have that depth of IA. Where folks were using these kinds of components, they were just switching content and didn't have a route attached to the views.

If each of those actually have a deep link url, that's actually very cool, in page navigation just doesn't support it right now.

I don't think you should double stack horizontal navs, but equally we shouldn't be using Tabs for navigation as it's completely the wrong semantic if each of those items do actually change the url when you action them.

We can take a dependency for a new variant, but I would be curious to hear about further use cases outside this section? If it's not super common, it will be difficult to prioritise due to the volume of other requests we have, which have much higher reuse. Do you happen to know how common this is?

**Reply by charleneyjh1** [reply]
Hi @SiTaggart, thanks for your response!

This is actually a very common use case across Segment. It appears in basically every Settings page for each of the products. 

Each of the vertical tabs currently in Segment has it's own URL. I can provide some screenshots if needed.

**Reply by charleneyjh1** [reply]
Besides in Unify settings, it also appears in every instances of a source settings page and destinations settings page, and privacy settings page. 
<img width="1480" alt="Screenshot 2023-07-06 at 1 09 07 PM" src="https://github.com/twilio-labs/paste/assets/56697620/a5717254-6a5b-45b0-9e9e-45efc97163a3">
<img width="1287" alt="Screenshot 2023-07-06 at 1 09 32 PM" src="https://github.com/twilio-labs/paste/assets/56697620/5c60a0f0-7046-4890-9a46-38b47e2d8c5e">
<img width="1381" alt="Screenshot 2023-07-06 at 1 10 02 PM" src="https://github.com/twilio-labs/paste/assets/56697620/90eb6331-daef-456b-9dc1-fa84d82b9f73">

**Comment by charleneyjh1** [comment]
Another thing that's related to in-page navigation is that Segment uses a lot of tertiary tabs with tables, and each of them have a different URL. (see screenshots) If we did the migration we will end up with stacking two horizontal in-page navigation again...
Curious to know if this was caught in the gap analysis for Evergreen-> Paste? 
<img width="1446" alt="Screenshot 2023-07-06 at 1 28 24 PM" src="https://github.com/twilio-labs/paste/assets/56697620/7069f72e-3299-45a4-99f4-4e13fe27d259">

**Reply by SiTaggart** [reply]
I suspect visually, when comparing components by components, everything is covered in isolation. For example, folks might have seen vertical tabs, considered them actual tabs, and called it done.

Same with horizontal tabs and in page navigation. On their own, the use cases are covered, but how they are implemented in the product might not have been fully considered by the Segment team at the time.

That is understandable, as there were a lot of things discovered once people started getting into the product. Things like undocumented custom components.

As these come up, we encourage the app foundations team to reevaluate what they need from Paste and file dependencies as they see fit, or continue creating custom components as has been the case previously, even working with Evergreen.

**Comment by loreina** [comment, marked as answer]
Hi, we have a ticket to fix this in our backlog. You can follow along here: https://twilio-productivity.atlassian.net/browse/DS-166.

**Reply by SiTaggart** [reply]
Vertical in page nav is released in Core `v20.5`

**Reply by charleneyjh1** [reply]
great news! I'll let my team know! Thank you so much
